src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java

```diff
@@ -51,17 +54,24 @@
 
     private final String colorMapName;
     private final String charset;
+    private final @Nonnull List<AnsiAttributeElement> openTags;
 
-    ColorConsoleAnnotator(String colorMapName, String charset) {
+    ColorConsoleAnnotator(String colorMapName, String charset, List<AnsiAttributeElement> openTags) {
         this.colorMapName = colorMapName;
         this.charset = charset;
-        LOGGER.fine("creating annotator with colorMapName=" + colorMapName + " charset=" + charset);
+        this.openTags = openTags;
+        LOGGER.log(Level.FINE, "creating annotator with colorMapName={0} charset={1}, openTags={2}", new Object[] { colorMapName, charset, openTags });
+    }
+
+    ColorConsoleAnnotator(String colorMapName, String charset) {
+        this(colorMapName, charset, Collections.emptyList());
     }
 
     @Override
     public ConsoleAnnotator<Object> annotate(Object context, MarkupText text) {
         String s = text.getText();
-        if (s.indexOf('\u001B') != -1) {
+        List<AnsiAttributeElement> nextOpenTags = openTags;
+        if (s.indexOf('\u001B') != -1 || !openTags.isEmpty()) {
```

Comment:
We could create a faster path for when `openTags.isEmpty && s.indexOf('\u001B') == -1`, since we'd just need to add tags at the beginning and end of the text without changing it. Not sure how important it is, but probably worth a comment in the source.

Reply:
Added in a95ea81.